cms/djangoapps/contentstore/tasks.py

```diff
@@ -1000,3 +1002,31 @@ def delete_v1_library(v1_library_key_string):
         "status": "SUCCESS",
         "msg": "SUCCESS"
     }
+
+
+@shared_task(time_limit=30)
+@set_code_owner_attribute
+def replace_all_library_source_blocks_ids_for_course(course, v1_to_v2_lib_map):
+    """Search a Modulestore for all library source blocks in a course by querying mongo.
+        replace all source_library_ids with the corresponding v2 value from the map
+    """
+    store = modulestore()
+    with store.bulk_operations(course.id):
+        for branch in [ModuleStoreEnum.BranchName.draft, ModuleStoreEnum.BranchName.published]:
+            blocks = store.get_items(
+                course.id.for_branch(branch),
+                settings={'source_library_id': {'$exists': True}}
+            )
+            for xblock in blocks:
+                try:
+                    xblock.source_library_id = v1_to_v2_lib_map[xblock.source_library_id]
+                    store.update_item(xblock, None)
```

Comment:
When this update is applied to a published version, does it create a new version that is not yet published? I don't know if this is the case, but ask in case this logic needs to add a publish operation on the newly created version in order to capture the intent that the published version now references v2 libraries in lieu of their v1 antecedents